### PR TITLE
net: Dummy L2 for offloaded ifaces

### PR DESCRIPTION
**Pull request by glarsennordic**
Adds dummy link layer for offloaded ifaces, allowing ifaces to directly receive l2_enable calls

Include extended version of `net_if_api`: `offloaded_if_api` to receive `l2_enable` calls.

### CODE REVIEW

**Comment by glarsennordic** [comment]
Split from: https://github.com/zephyrproject-rtos/zephyr/pull/53118
In service of: https://github.com/zephyrproject-rtos/zephyr/issues/52695

**Review by rlubos — APPROVED** [review]
LGTM

File: tests/net/all/src/main.c [hunk 1]
```diff
@@ -18,23 +18,31 @@ LOG_MODULE_REGISTER(net_test, LOG_LEVEL_DBG);
 #include <zephyr/net/net_if.h>
```
Comment (jukkar) [inline tests/net/all/src/main.c:18]: Basically LTGM. Small nit about the tests, was the idea that the tests would be in the second commit? Now the first commit contains also some test code.

Comment (glarsennordic) [reply tests/net/all/src/main.c:18]: Yes, that was the idea

The changes to `tests/net/all` and `tests/net/socket/offload_dispatcher` are just minor tweaks to ensure both tests still build and run successfully with the new changes, whereas the second test introduces an entire new test suite

The reason I added a second net device to `tests/net/all` is that originally that test was using a net device that was simultaneously `offloaded` and `dummy`. It is no longer possible to have a device be both simultaneously, so I instanced one of each

**Review by nandojve — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**